### PR TITLE
feat: import read-only wallet by ens

### PR DESCRIPTION
**Pull request by izayl**
image
image

### CODE REVIEW

**Comment by izayl** [comment]
related issue: https://github.com/tallycash/extension/pull/435

**Comment by mhluongo** [comment]
Thanks for the contribution @izayl! Reviewing it now :smile:

File: ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx [hunk 1]
```diff
@@ -17,8 +20,14 @@ export default function OnboardingViewOnlyWallet(): ReactElement {
 
   const handleSubmitViewOnlyAddress = useCallback(async () => {
     const trimmedAddress = address.trim()
-
-    if (isAddress(trimmedAddress)) {
+    if (trimmedAddress.endsWith(".eth")) {
```
Comment (mhluongo) [inline ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx:23]: ENS supports more than just "*.eth", but I think it's all we resolve rn — I'll open an issue for other TLDs

File: ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx [hunk 2]
```diff
@@ -51,7 +60,8 @@ export default function OnboardingViewOnlyWallet(): ReactElement {
       <div className="content">
         <h1 className="serif_header">Explore Tally</h1>
         <div className="subtitle">
-          Add an Ethereum address to view an existing wallet in Tally.
+          Add an Ethereum address or ENS domain to view an existing wallet in
```
Comment (mhluongo) [inline ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx:63]: I prefer "ENS name" to "ENS domain" but can revise the copy in another PR

**Review by mhluongo — APPROVED** [review]
A couple nits, but if everything is green I think we merge this as is. Impressive first PR @izayl!

**Comment by mhluongo** [comment]
Manually merging here. CI is choking on a missing env variable, opened issue #793

**Pull request merged**